Jupyter magics v2 (#846)
DAGWorks-Inc/hamilton: examples/jupyter_notebook_magic/README.md

# Hamilton notebook extension

> One of the best part about notebooks is the ability to execute and immediately inspect results. They provide a "read-eval-print" loop (REPL) coding experience. However, the way Hamilton separates dataflow definition (functions in a module) from execution (building and executing a driver) creates an extra step that can slowdown this loop.

Reviewer: ```suggestion
One of the best part about notebooks is the ability to execute and immediately inspect results. They provide a "read-eval-print" loop (REPL) coding experience. However, the way Hamilton separates dataflow definition (functions in a module) from execution (building and executing a driver) creates an extra step that creates a little friction with this loop.
```

One of the best part about notebooks is the ability to execute and immediately inspect results. They provide a "read-eval-print" loop (REPL) coding experience. However, the way Hamilton separates dataflow definition (functions in a module) from execution (building and executing a driver) creates an extra step that can slowdown this loop.

We built the Hamilton notebook extension to tighten that loop and even give a better experience than the core notebook experience!

To load the magic:
```python
# load some extensions / magic...
%load_ext hamilton.plugins.jupyter_magic
```

Then to use it:
For example, this would allow you to define the module `joke` from your notebook

```python

> %%cell_to_module -m MODULE_NAME # more args

Reviewer: should have output of `?cell_to_module` instead.
